engine/src/main/java/org/terasology/logic/characters/CharacterMovementSystemUtility.java

```diff
@@ -101,23 +98,35 @@ public void setToExtrapolateState(EntityRef entity, CharacterStateEvent state, l
         Vector3f newPos = new Vector3f(state.getVelocity());
         newPos.scale(t);
         newPos.add(state.getPosition());
-        LocationComponent location = entity.getComponent(LocationComponent.class);
+        extrapolateLocationComponent(entity, state, newPos);
+
+        extrapolateCharacterMovementComponent(entity, state);
+
+        extrapolateCharacterComponent(entity, state);
+        setPhysicsLocation(entity, newPos);
+    }
+
+	private void extrapolateLocationComponent(EntityRef entity, CharacterStateEvent state, Vector3f newPos) {
+		LocationComponent location = entity.getComponent(LocationComponent.class);
```

Comment:
It looks like you're using tab characters instead of spaces. Mind fixing? We use 4 spaces instead of tab (Checkstyle will catch these)